tests/core/data/maxwellian_particle_initializer/test_init_funcs.h

```diff
@@ -0,0 +1,368 @@
+
+
+    for (std::uint32_t ix = ix0; ix < ix1; ++ix)
+    {
+        for (std::uint32_t iy = iy0; iy < iy1; ++iy)
+        {
+            for (std::uint32_t iz = iz0; iz < iz1; ++iz)
+            {
+                double n; // cell centered density
+                std::array<double, 3> particleVelocity;
+                std::array<std::array<double, 3>, 3> basis;
+
+                // get the coordinate of the current cell
+                auto coord = layout.cellCenteredCoordinates(ix, iy, iz);
+                auto x     = coord[0];
+                auto y     = coord[1];
+                auto z     = coord[2];
+
+                // now get density, velocity and thermal speed values
+                n         = density_(x, y, z);
+                auto Vx   = vx(x, y, z);
+                auto Vy   = vy(x, y, z);
+                auto Vz   = vz(x, y, z);
+                auto Vthx = vthx(x, y, z);
+                auto Vthy = vthy(x, y, z);
+                auto Vthz = vthz(x, y, z);
+
+                // weight for all particles in this cell
+                auto cellWeight = n * cellVolume / nbrParticlePerCell_;
+
+                ParticleDeltaDistribution randPosX;
+                ParticleDeltaDistribution randPosY;
+                ParticleDeltaDistribution randPosZ;
+
+                if (basis_ == Basis::Magnetic)
+                {
+                    auto Bx = bx(x, y, z);
+                    auto By = by(x, y, z);
+                    auto Bz = bz(x, y, z);
+
+                    localMagneticBasis({Bx, By, Bz}, basis);
+                }
+
+                for (std::uint32_t ipart = 0; ipart < nbrParticlePerCell_; ++ipart)
+                {
+                    maxwellianVelocity({Vx, Vy, Vz}, {Vthx, Vthy, Vthz}, generator,
+                                       particleVelocity);
+
+                    if (basis_ == Basis::Magnetic)
+                    {
+                        particleVelocity = basisTransform(basis, particleVelocity);
+                    }
+
+                    std::array<float, dimension> delta
+                        = {{randPosX(generator), randPosY(generator), randPosZ(generator)}};
+
+
+                    Particle<dimension> tmpParticle;
+
+                    // particle iCell is in AMR index
+                    auto AMRCellIndex = layout.localToAMR(Point{ix, iy, iz});
+
+                    tmpParticle.weight = cellWeight;
+                    tmpParticle.charge = particleCharge_;
+                    tmpParticle.iCell  = AMRCellIndex.template toArray<int>();
+                    tmpParticle.delta  = delta;
+                    tmpParticle.v      = particleVelocity;
+
+                    particles.push_back(std::move(tmpParticle));
+                } // end particle looop
+            }     // end z
+        }         // end y
+    }             // end x
+}
+
+
+
+} // namespace PHARE::core::test_fn::func_3d
+
+
+#endif // PHARE_TEST_MAXWELLIAN_PARTICLE_INITIALIZER_TEST_INIT_FUNCS_H
```

Comment:
is that the former code that's tested to ensure the new one gives the same?

Reply:
> is that the former code
yes
> to ensure the new one gives the same
something like that, not done but could be

Reply:
would you like this to be considered in this PR or will I make an issue?